### PR TITLE
feat: [CDS-84377]: Add examples of using output variables within step group

### PR DESCRIPTION
**Pull request by krishi0408**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md [hunk 1]
```diff
@@ -78,10 +78,17 @@ You can refer step group variables within the step group using the expression, `
 
 You can refer step group variables outside the step group using the expression, `<+pipeline.stages.[stage Id].spec.execution.steps.[step group id].variables.[variable name]>`.
 
-:::info
+### Output variables
 
-Execution input is not supported for step group variables.
+You can refer to output variables using expression variable expression which requires the step group name to be present, `<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`
+
+If you are accessing the output variable in a steo within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`
```
Comment (SushrutHarness) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md:85]:
```suggestion
If you are accessing the output variable in a step within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`.
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md [hunk 2]
```diff
@@ -78,10 +78,17 @@ You can refer step group variables within the step group using the expression, `
 
 You can refer step group variables outside the step group using the expression, `<+pipeline.stages.[stage Id].spec.execution.steps.[step group id].variables.[variable name]>`.
 
-:::info
+### Output variables
 
-Execution input is not supported for step group variables.
+You can refer to output variables using expression variable expression which requires the step group name to be present, `<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`
```
Comment (SushrutHarness) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md:83]: I'm kinda confused on what expression variable expression means?

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md [hunk 3]
```diff
@@ -78,10 +78,17 @@ You can refer step group variables within the step group using the expression, `
...
+
+If you are accessing the output variable in a steo within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`
+
+If your step group is configured as loop, there is an index added at the end of identifier to identify the unique loop instance you can refer the output variable in this case as, `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`
```
Comment (SushrutHarness) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md:87]:
```suggestion
If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance you can use to refer to the output variable. In this case: `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`
```

**Review by SushrutHarness — COMMENTED** [review]
A question, and some formatting suggestions.

**Review by acasian-harness — DISMISSED** [review]

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md [hunk 4]
```diff
@@ -78,10 +78,17 @@ You can refer step group variables within the step group using the expression, `
 
 You can refer step group variables outside the step group using the expression, `<+pipeline.stages.[stage Id].spec.execution.steps.[step group id].variables.[variable name]>`.
 
-:::info
+### Output variables
 
-Execution input is not supported for step group variables.
+You can refer to output variables using expression variable in expression which requires the step group name to be present, `<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md:83]:
```suggestion
You can reference step group output variables using expressions using the step group name. Here's a sample expression:
`<+execution.steps.<stepGroupID>.steps.stepID.outputVariables.[variable name]>`.
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md [hunk 5]
```diff
@@ -78,10 +78,17 @@ You can refer step group variables within the step group using the expression, `
...
+
+If you are accessing the output variable in a step within the same step group you can use a shorter notation, `<+stepGroup.steps.step1Identifier.output.outputVariables.[variable name]>`.
+
+If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance you can use to refer to the output variable. In this case: `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/step-groups.md:87]:
```suggestion
If your step group is configured as a loop, there is an index added at the end of the identifier to identify the unique loop instance. You can use this unique loop instance to reference the output variables. For example, `<+execution.steps.<stepGroupID_[index]>.steps.stepID.outputVariables.[variable name]>`.
```

**Review by roshnisarangadharan — APPROVED** [review]
Approving with the assumption that you will make the suggested updates.

**Pull request merged**